go/libkb/device_clone_token.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package libkb
+
+import "fmt"
+
+const (
+	DefaultCloneTokenValue string = "00000000000000000000000000000000"
+)
+
+type DeviceCloneToken struct {
+	Prior  string
+	Stage  string
+	Clones int
+}
+
+func deviceCloneTokenPaths(un NormalizedUsername) (p, s, c string) {
+	basePath := fmt.Sprintf("users.%s.device_clone_token", un)
+	p = fmt.Sprintf("%s.prior", basePath)
+	s = fmt.Sprintf("%s.stage", basePath)
+	c = fmt.Sprintf("%s.clones", basePath)
+	return
+}
+
+func (f JSONConfigFile) GetDeviceCloneToken(un NormalizedUsername) DeviceCloneToken {
+	pPath, sPath, cPath := deviceCloneTokenPaths(un)
+	p, _ := f.GetStringAtPath(pPath)
+	s, _ := f.GetStringAtPath(sPath)
+	c, _ := f.GetIntAtPath(cPath)
+	return DeviceCloneToken{Prior: p, Stage: s, Clones: c}
+}
+
+func (f *JSONConfigFile) SetDeviceCloneToken(un NormalizedUsername, d DeviceCloneToken) error {
```

Comment:
Let's set these atomically. Is it crash safe if saved in this order?

Reply:
i'm 95% sure that this could fail at any point and we'd be ok. the worst case scenario for crashing in the middle and the next run completing is simply that the notification would be delayed until then. i'll take a look at doing them atomically though. i'd rather future me (and anyone else) not have to figure that out again.